### PR TITLE
Sync client-go – third try

### PR DESCRIPTION
**Pull request by sttts**
Follow-up of https://github.com/kubernetes/kubernetes/pull/40692 & https://github.com/kubernetes/kubernetes/pull/40699.

Approval and lgtm from https://github.com/kubernetes/kubernetes/pull/40692.

### CODE REVIEW

File: hack/make-rules/verify.sh [hunk 1]
```diff
@@ -23,10 +23,9 @@ source "${KUBE_ROOT}/cluster/lib/util.sh"
 
 # Excluded checks are always skipped.
 EXCLUDED_CHECKS=(
-  "verify-linkcheck.sh"  # runs in separate Jenkins job once per day due to high network usage
-  "verify-govet.sh"      # it has a separate make vet target
-  "verify-staging-client-go.sh" # TODO: enable the script after 1.5 code freeze
```
Comment (deads2k) [inline hack/make-rules/verify.sh]: thsi has to stay excluded.  cycle

**Comment by deads2k** [comment]
`client-go: initial examples import` - the intent is good, but my guess is that this will make a mess of whatever they're using to sync.

File: hack/verify-staging-client-go.sh [hunk 2]
```diff
@@ -72,3 +72,16 @@ godep restore ${V} 2>&1 | sed 's/^/  /'
 
 echo "Testing staging/copy.sh"
 staging/copy.sh -d 2>&1 | sed 's/^/  /'
+
+# Check that the apimachinery Godeps revision is actually in Github
+echo "Checking that a valid k8s.io/apimachinery is reference in client-go"
```
Comment (deads2k) [inline hack/verify-staging-client-go.sh]: unnecessary. This is always the case unless you do something insane and manual to your godeps.

Comment (sttts) [inline hack/verify-staging-client-go.sh]: That's the point: due to the cycle during development we will do insane things.

File: hack/update-client-go.sh [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  esac
+done
+readonly V
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+cd ${KUBE_ROOT}
+
+# Create a temporary GOPATH for apimachinery and client-go, copy the current master into each and turn them
+# into a git repo. Then we can run copy.sh with this additional GOPATH. The Godeps.json will
+# have invalid git SHA1s, but it's good enough as a smoke test of copy.sh.
```
Comment (sttts) [inline hack/update-client-go.sh]: To make sure copy.sh is working?

Comment (sttts) [inline hack/update-client-go.sh]: how do you run godep without a real repo?

File: hack/verify-staging-client-go.sh (hunk 2 above)
Comment (sttts) [inline hack/verify-staging-client-go.sh]: The idea: `hack/update-client-go.sh -s`.

This gives you a client which has not valid apimachinery upstream rev, but works great for WIP PRs.

**Comment by deads2k** [comment]
@k8s-bot kubemark e2e test this

**Pull request merged**